src/compiler/parser.ts

```diff
@@ -5889,9 +5891,13 @@ namespace ts {
                     case SyntaxKind.SymbolKeyword:
                     case SyntaxKind.VoidKeyword:
                         return parseTokenNode<JSDocType>();
+                    case SyntaxKind.StringLiteral:
+                    case SyntaxKind.NumericLiteral:
+                    case SyntaxKind.TrueKeyword:
+                    case SyntaxKind.FalseKeyword:

```

Comment:
Actually, those are not literal types, it turns out. In the interests of getting this PR in, I'll do those separately.